DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/CryptoHashProcessor.cs

```diff
@@ -0,0 +1,88 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Text;
+using Dicom;
+using Dicom.IO.Buffer;
+using EnsureThat;
+using Microsoft.Health.DeID.SharedLib.Settings;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Dicom.DeID.SharedLib;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors
+{
+    public class CryptoHashProcessor : IAnonymizerProcessor
+    {
+        private readonly CryptoHashFunction _cryptoHashFunction;
+
+        public CryptoHashProcessor(JObject settingObject, IAnonymizerSettingsFactory settingFactory = null)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            settingFactory ??= new AnonymizerSettingsFactory();
```

Comment:
>settingFactory ??= new AnonymizerSettingsFactory(); [](http://example.com/codeflow?start=12&length=51)

I really don't like checking the settingFactory here. If you are not setting this parameter in its caller, you may create multiple SettingFactory instances. So I would like to that in two ways:

1) I check this object in RuleFactory level and create a new object there, so in these processors, we jusr check if it is NULL like the settingObject
2) I remove this parameter in this function and create it in every processor. This may create multiple instances but the interface is better. After all this settingFactory is not used in RuleFactory, right?

I think both of them are OK, and personally I prefer the second one to make the interface clean :-)

Reply:
Thanks a lot, I use the second method to avoid this parameter.